main/api.c

```diff
@@ -374,25 +417,9 @@ int api_socket_start(struct event_base *base) {
 		return errno_log(errno, "socket");
 
 	addr.un.sun_family = AF_UNIX;
-	memccpy(addr.un.sun_path, path, 0, sizeof(addr.un.sun_path) - 1);
-
-	ret = bind(fd, &addr.a, sizeof(addr.un));
-	if (ret < 0 && errno == EADDRINUSE) {
-		// unix socket file exists, check if there is a process
-		// listening on the other side.
-		ret = connect(fd, &addr.a, sizeof(addr.un));
-		if (ret == 0) {
-			LOG(ERR, "grout already running on API socket %s, exiting", path);
-			close(fd);
-			return errno_set(EADDRINUSE);
-		}
-		if (ret < 0 && errno != ECONNREFUSED)
-			return errno_log(errno, "connect");
-		// remove socket file, and try to bind again
-		if (unlink(addr.un.sun_path) < 0)
-			return errno_log(errno, "unlink");
-		ret = bind(fd, &addr.a, sizeof(addr.un));
-	}
+	memcpy(addr.un.sun_path, GR_SOCK_NAME, GR_SOCK_NAME_LEN);
+
+	ret = bind(fd, &addr.a, offsetof(struct sockaddr_un, sun_path) + GR_SOCK_NAME_LEN);
```

Comment:
I had a look at https://man7.org/linux/man-pages/man7/unix.7.html and it seems you should rewrite it as:

```c
addr.un.sun_path[0] = '\0';
memccpy(addr.un.sun_path + 1, gr_config.api_sock_name, 0, sizeof(addr.un.sun_path) - 2);

ret = bind(fd, &addr.a, sizeof(addr.un));
```